Move filter specific code from BitmapTextureGL to TextureMapperGL.

Aki-7 opened the pull request:
#### 499bdf98d4a7547259c7ed0af2486f4dff20dbd6
<pre>
Move filter specific code from BitmapTextureGL to TextureMapperGL.
<a href="https://bugs.webkit.org/show_bug.cgi?id=261022">https://bugs.webkit.org/show_bug.cgi?id=261022</a>

Reviewed by Fujii Hironori.

This is a preparation refactoring for <a href="https://bugs.webkit.org/show_bug.cgi?id=231653.">https://bugs.webkit.org/show_bug.cgi?id=231653.</a>

To fix the issue above, we have to change the number of passes needed to
do blur filtering according to the standard deviation of the blur.
Now the number of passes for each filter is specified in
BitmapTextureGL.cpp but other parameters regarding filtering such as
shader options are specified in TextureMapperGL.cpp, which leads to the
inconsistency and low readability.

This change moves filter specific code from BitmapTextureGL.cpp
to TextureMapperGL.cpp

* Source/WebCore/platform/graphics/texmap/BitmapTextureGL.cpp:
(WebCore::BitmapTextureGL::applyFilters):
(WebCore::getPassesRequiredForFilter): Deleted.
* Source/WebCore/platform/graphics/texmap/BitmapTextureGL.h:
(WebCore::BitmapTextureGL::FilterInfo::FilterInfo):
(WebCore::BitmapTextureGL::setFilterInfo):
* Source/WebCore/platform/graphics/texmap/TextureMapperGL.cpp:
(WebCore::TextureMapperGL::drawTexture):
(WebCore::TextureMapperGL::drawTexturePlanarYUV):
(WebCore::TextureMapperGL::drawTextureSemiPlanarYUV):
(WebCore::TextureMapperGL::drawTexturePackedYUV):
(WebCore::TextureMapperGL::drawFilterPass):
(WebCore::getPassesRequiredForFilter):
(WebCore::TextureMapperGL::applyFilter):
(WebCore::TextureMapperGL::drawFiltered): Deleted.
* Source/WebCore/platform/graphics/texmap/TextureMapperGL.h:
* Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp:
(WebCore::TextureMapperLayer::paintIntoSurface):

Canonical link: <a href="https://commits.webkit.org/267599@main">https://commits.webkit.org/267599@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a56bd9c16a0335f201c1ac5b3659c8eb35b198f4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17127 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17452 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17943 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18913 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16025 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/17319 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20726 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17594 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/18224 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17330 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/17679 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14864 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19729 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/14919 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15555 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/22245 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/15918 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15722 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/20062 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16313 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13834 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15467 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19834 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2104 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16144 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Aki-7 commented on Source/WebCore/platform/graphics/texmap/BitmapTextureGL.cpp:
```diff
@@ -162,66 +162,26 @@ void BitmapTextureGL::updateContents(Image* image, const IntRect& targetRect, co
-    RefPtr<BitmapTexture> spareSurface;
-
-    m_filterInfo = FilterInfo();
+    RefPtr<BitmapTexture> surface = this;
```
The CI pointed out that here is a style error.
But according to the [code style guideline](https://webkit.org/code-style-guidelines/#names-protectors-this), "Ref and RefPtr objects meant to protect this from deletion should be named “protectedThis”.", and I don't think it's applicable here.

Should I address all the CI errors, or can we proceed with some if they’re reasonable?

fujii replied:
You can ignore it.

fujii reviewed (changes requested):
You should wrap lines of the commit message.

Commit:
Move filter specific code from BitmapTextureGL to TextureMapperGL.
https://bugs.webkit.org/show_bug.cgi?id=261022

Reviewed by Fujii Hironori.

This is a preparation refactoring for https://bugs.webkit.org/show_bug.cgi?id=231653.

To fix the issue above, we have to change the number of passes needed to
do blur filtering according to the standard deviation of the blur.
Now the number of passes for each filter is specified in
BitmapTextureGL.cpp but other parameters regarding filtering such as
shader options are specified in TextureMapperGL.cpp, which leads to the
inconsistency and low readability.

This change moves filter specific code from BitmapTextureGL.cpp
to TextureMapperGL.cpp

* Source/WebCore/platform/graphics/texmap/BitmapTextureGL.cpp:
(WebCore::BitmapTextureGL::applyFilters):
(WebCore::getPassesRequiredForFilter): Deleted.
* Source/WebCore/platform/graphics/texmap/BitmapTextureGL.h:
(WebCore::BitmapTextureGL::FilterInfo::FilterInfo):
(WebCore::BitmapTextureGL::setFilterInfo):
* Source/WebCore/platform/graphics/texmap/TextureMapperGL.cpp:
(WebCore::TextureMapperGL::drawTexture):
(WebCore::TextureMapperGL::drawTexturePlanarYUV):
(WebCore::TextureMapperGL::drawTextureSemiPlanarYUV):
(WebCore::TextureMapperGL::drawTexturePackedYUV):
(WebCore::TextureMapperGL::drawFilterPass):
(WebCore::getPassesRequiredForFilter):
(WebCore::TextureMapperGL::applyFilter):
(WebCore::TextureMapperGL::drawFiltered): Deleted.
* Source/WebCore/platform/graphics/texmap/TextureMapperGL.h:
* Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp:
(WebCore::TextureMapperLayer::paintIntoSurface):

Canonical link: https://commits.webkit.org/267599@main
